Update 2025

reingart opened the pull request:
## Summary

Actualizaciones acumuladas 2025:

* Factura Electronica WSFEv1 Version 4: cancela_misma_moneda_ext & condicion_iva_receptor_id  RG5616/24
* Facturas B Ley 27743: IVA contenido (PDF)
* Actvidades Agropecuarias RG 5259
* Otros webservices: remito harinero y Carta de Porte

Incluye ajustes a Python3 y resolucion de conflictos.
Pendientes:

* revison unicode 
* test unitarios
* ajustes finales plantilla PDF

## Checklist

- [ ] Classes, Variables, function and methods logic  ok
- [ ] Comments written explaining what the code does
- [ ] All python code is PEP8 compliant (run black .)
- [ ] No lint issues (run flake8)
- [ ] Test coverage with pytest implemented
- [ ] Reviewers assigned (at least 1 mentor)

## Manual test evidence

(attach command-line examples, execution output & logs, etc.)

reingart committed:
RECEX1: registro A: actividades RG5259/22 RG5264/22 FEv3

reingart committed:
WSRemHarina: agrego ejemplo Visual Basic Script

reingart committed:
WSCPE: verion AFIP 2.0.3

 * EditarCPEConfirmadaAutomotorDG
 * EditarCPEConfirmadaFerroviariaDG
 * EditarCPEAutomotor (cuit_pagador_flete obligatorio)
 * EditarCPEAutomotor (se agrega tarifa, km_recorrer, observaciones)

reingart committed:
WSCPE: ajuste editar cpe

reingart committed:
PyFEPDF: IVA Ley 27743

reingart committed:
PyFEPDF: ajustes iva consumidor final

reingart committed:
PyFEPDF: exento para discriminar iva

reingart committed:
WSFEv1: Version 4 Nuevos campos FECAEADetRequest

* <CanMisMonExt> S o N  cancela_misma_moneda_ext
* <CondicionIVAReceptorId> int condicion_iva_receptor_id

reingart committed:
RECE1: actividades

reingart committed:
RECE1: Version 4 nuevos campos

* cancela_misma_moneda_ext
* condicion_iva_receptor_id

reingart committed:
WSFEv1: Version 4 FEParamGetCondicionIvaReceptor

reingart committed:
WSFEv1: categoria de iva opcional

reingart committed:
WSFEv1: agrego fecha al consultar cotizacion

dcrocetti commented:
Buenas tardes Romina/Mariano.
Cuando intenté compilar con los nuevos cambios pyfepdf.py dio error de identación en la línea 1341. Corrí la línea como está debajo y se solucionó. (No sé si es el lugar correcto para enviar estos casos.)

1338                                                if p == int(p):
1339                                                    imp_it_bruto = importe_it / (Decimal(p) / Decimal(100) + Decimal(1))
1340                                                    imp_it_iva = importe_it - imp_it_bruto
1341                                                    iva_liq += imp_it_iva

estudiosistemas commented:
Hola Mariano/Romina!

Les dejo algunas consideraciones que tuve al generar el instalador. siguiendo los pasos que están en actions.

Como referencia, la instalación la hice en dos maquinas, un Win11 de 64bits y un Win 10 de 32 bits, y en las dos me dio los mismos errores.

1. Al hacer "python setup.py install" me dio un error de "deprecationWarning: setup.py install is deprecated."
    - lo solucione instalando una version anterior de setuptools "pip install setuptools==58.2.0"

2.  Al incluir import pyi25 y/o from pyafipws import pyqr en setup_win.py y hacer  python setup_win.py py2exe, me dio el error "no module named pil.imagetk".
    - lo soluciones quitando las referencias _imagingtk", "PIL._imagingtk", "ImageTk", "PIL.ImageTk" de la linea 149 en excludes

    excludes=["pywin", "pywin.dialogs", "pywin.dialogs.list", "win32ui",
            "Tkconstants","Tkinter","tcl",
            "_imagingtk", "PIL._imagingtk", "ImageTk", "PIL.ImageTk", "FixTk",
            ]

3. Una vez generado el instalador, al instalar me daba el error : "UnboundLocalError: local variable 'url' referenced before assignmen".
    - esto lo solucione agregando "url = None" en def main(): del archivo pyqr.py

4. El error que no pude solucionar es el que da al instanciar el modulo pyqr en vb6,  "Set pyqr = CreateObject("PyQR")" 

Traceback (most recent call last):
  File "win32com\server\policy.pyc", line 151, in CreateInstance
  File "win32com\server\policy.pyc", line 215, in _CreateInstance_
  File "win32com\server\policy.pyc", line 803, in call_func
  File "win32com\server\policy.pyc", line 791, in resolve_func
  File "win32com\server\policy.pyc", line 812, in _import_module
ModuleNotFoundError: No module named 'pyqr_com'
pythoncom error: CPyFactory::CreateInstance failed to create instance. (80004005)
pythoncom error: ERROR: server.policy could not create an instance.

espero poder ser de ayuda y gracias por todo el trabajo que hacen!!!

reingart committed:
Merge branch 'main' into 2025

reingart committed:
Merge branch 'main' into 2025

reingart committed:
PyFEPDF: fix indent merge conflict

reingart committed:
PyQR: fix win installer deps

Fix error "no module named pil.imagetk"
Thanks @estudiosistemas!

reingart committed:
PyQR: fix return vallue

Fixes ""UnboundLocalError: local variable 'url' referenced before assignment"
Thanks @estudiosistemas

reingart committed:
Merge branch 'main' into 2025

reingart committed:
WSFEv1: disable parameters UT

Temporarily disable unit test until new snapshots of webserver communications are taken

```
=========================== short test summary info ============================
FAILED tests/test_wsfev1.py::test_main_parametros - RuntimeError: Operation FEParamGetCondicionIvaReceptor not found in WSDL: Service/Port Type: ('Service', 'ServiceSoap')
```

reingart commented:
Gracias @estudiosistemas por los comentarios!, ya estarian los ajustes (tambien pueden usar directamente la herramenienta de sugerencia para dejar una modificacion a los archivos)

Sobre el tema de pyqr, podrias probar ejecutando `pyqr.exe --register`? , deberia funcionarles luego.
Si el problema persiste, sugiero abrir otro ticket, porque no ha cambiado el QR y es un tema complementario (pueden seguir usando la version anterior).

Ya actualice todo el flujo de GiHub Actions para compilar, probar y generar el instalador automaticamente; pueden probar el que se prepara en los artefactos: https://github.com/reingart/pyafipws/pull/120/checks
![image](https://github.com/user-attachments/assets/f904bb5e-c15b-4e6d-a468-3e149eb5f553)

estudiosistemas commented:
Hola Mariano! Me alegra haber podido colaborar un poquito a todo el laburo que hacen! Voy a compilar todo de nuevo y probar el registro de pyqr para ver si funciona instanciado el objeto desde vb6 y te aviso. Abrazo!

dcrocetti commented:
Buen día Mariano, buen día al grupo completo ! 

Adhiero al agradecimiento de los demás integrantes. Realmente me gustaría colaborar en el proyecto más activamente. Reconozco limitaciones en mis conocimientos para poder hacerlo de esa manera, De todas formas estoy incursionando en Python. Quedé obsoleto con las nuevas tecnologías, es así. 

Contarles además que compilé con los nuevos archivos y todo salió OK. 

Gracias Gracias ! ! !

consertec commented:
> Buen día Mariano, buen día al grupo completo !
> 
> Adhiero al agradecimiento de los demás integrantes. Realmente me gustaría colaborar en el proyecto más activamente. Reconozco limitaciones en mis conocimientos para poder hacerlo de esa manera, De todas formas estoy incursionando en Python. Quedé obsoleto con las nuevas tecnologías, es así.
> 
> Contarles además que compilé con los nuevos archivos y todo salió OK.
> 
> Gracias Gracias ! ! !

Hola! Una consulta:
Compilaste descargando la rama "2025" y siguiendo los pasos del Workflow de Actions?

dcrocetti commented:
> > Buen día Mariano, buen día al grupo completo !
> > Adhiero al agradecimiento de los demás integrantes. Realmente me gustaría colaborar en el proyecto más activamente. Reconozco limitaciones en mis conocimientos para poder hacerlo de esa manera, De todas formas estoy incursionando en Python. Quedé obsoleto con las nuevas tecnologías, es así.
> > Contarles además que compilé con los nuevos archivos y todo salió OK.
> > Gracias Gracias ! ! !
> 
> Hola! Una consulta: Compilaste descargando la rama "2025" y siguiendo los pasos del Workflow de Actions?

Hola, Buenas tardes! 
Tengo una instalación vieja en Python 2.7 donde compilo sin problemas. Lo que hice fue descargar los 6 u 7 archivos modificados y compilé sin inconvenientes. 
Tengo pendiente hacer en otra instalación o en Entorno Virtual las pruebas con Python 3 siguiendo el Workflow de actions.

consertec commented:
Hola!
Sabes que estoy intentando lo mismo hace 4 o 5 días! Pero no he logrado
compilar aún. Creía que eran solo dos archivos (wsaa y wsfev1). Algo le
estoy pifiando. Los modificados los bajaste de la rama 2025?

El mar, 4 de mar de 2025, 6:44 p. m., dcrocetti ***@***.***>
escribió:

> Buen día Mariano, buen día al grupo completo !
> Adhiero al agradecimiento de los demás integrantes. Realmente me gustaría
> colaborar en el proyecto más activamente. Reconozco limitaciones en mis
> conocimientos para poder hacerlo de esa manera, De todas formas estoy
> incursionando en Python. Quedé obsoleto con las nuevas tecnologías, es así.
> Contarles además que compilé con los nuevos archivos y todo salió OK.
> Gracias Gracias ! ! !
>
> Hola! Una consulta: Compilaste descargando la rama "2025" y siguiendo los
> pasos del Workflow de Actions?
>
> Hola, Buenas tardes!
> Tengo una instalación vieja en Python 2.7 donde compilo sin problemas. Lo
> que hice fue descargar los 6 u 7 archivos modificados y compilé sin
> inconvenientes.
> Tengo pendiente hacer en otra instalación o en Entorno Virtual las pruebas
> con Python 3 siguiendo el Workflow de actions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699007703>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEGXKTUXUXHEWYIO5ID27L2SYNDNAVCNFSM6AAAAABXVIN25OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJZGAYDONZQGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
> [image: dcrocetti]*dcrocetti* left a comment (reingart/pyafipws#120)
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699007703>
>
> Buen día Mariano, buen día al grupo completo !
> Adhiero al agradecimiento de los demás integrantes. Realmente me gustaría
> colaborar en el proyecto más activamente. Reconozco limitaciones en mis
> conocimientos para poder hacerlo de esa manera, De todas formas estoy
> incursionando en Python. Quedé obsoleto con las nuevas tecnologías, es así.
> Contarles además que compilé con los nuevos archivos y todo salió OK.
> Gracias Gracias ! ! !
>
> Hola! Una consulta: Compilaste descargando la rama "2025" y siguiendo los
> pasos del Workflow de Actions?
>
> Hola, Buenas tardes!
> Tengo una instalación vieja en Python 2.7 donde compilo sin problemas. Lo
> que hice fue descargar los 6 u 7 archivos modificados y compilé sin
> inconvenientes.
> Tengo pendiente hacer en otra instalación o en Entorno Virtual las pruebas
> con Python 3 siguiendo el Workflow de actions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699007703>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEGXKTUXUXHEWYIO5ID27L2SYNDNAVCNFSM6AAAAABXVIN25OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJZGAYDONZQGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

dcrocetti commented:
Si..
bajé los archivos de:
https://github.com/reingart/pyafipws/pull/120/files
(hubo varios paquetes que se actualizaron (requirements.txt).
python setup.py py2exe (así compila bien y funciona correctamente)

Con:
python setup_win.py py2exe (compila bien, pero al momento de instanciar los objetos desde vb6 me abre una ventana de python y se me cuelga el sistema)

cperezabo commented:
¿Hay un ETA para que esto esté en main?

consertec commented:
Una consulta. Entonces bajastes los archivos modificados, luego los pusiste
en la carpeta de pyafipws, y compilaste con python setup.py py2exe. Ahora
que requirements ejecutaste antes de compilar? Hice todo pero no obtengo un
exe.

El mar, 4 de mar de 2025, 9:01 p. m., dcrocetti ***@***.***>
escribió:

> Si..
> bajé los archivos de:
> https://github.com/reingart/pyafipws/pull/120/files
> (hubo varios paquetes que se actualizaron (requirements.txt).
> python setup.py py2exe (así compila bien y funciona correctamente)
>
> Con:
> python setup_win.py py2exe (compila bien, pero al momento de instanciar
> los objetos desde vb6 me abre una ventana de python y se me cuelga el
> sistema)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699304437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEGXKSDQ2NHTRRRDW5FFHL2SY5GXAVCNFSM6AAAAABXVIN25OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJZGMYDINBTG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
> [image: dcrocetti]*dcrocetti* left a comment (reingart/pyafipws#120)
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699304437>
>
> Si..
> bajé los archivos de:
> https://github.com/reingart/pyafipws/pull/120/files
> (hubo varios paquetes que se actualizaron (requirements.txt).
> python setup.py py2exe (así compila bien y funciona correctamente)
>
> Con:
> python setup_win.py py2exe (compila bien, pero al momento de instanciar
> los objetos desde vb6 me abre una ventana de python y se me cuelga el
> sistema)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reingart/pyafipws/pull/120#issuecomment-2699304437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEGXKSDQ2NHTRRRDW5FFHL2SY5GXAVCNFSM6AAAAABXVIN25OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJZGMYDINBTG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

pbordon commented:
Probé con pyqr.exe --register  pyqr_com.exe --register y nada, en VB6 da error al Crear el objeto.
Como decís vos Mariano, no cambió nada en el pyqr.
Estoy compilando todo con python 3.10 , a través de python setup_win.py py2exe.
Hubo algún fix mas?
Gracias

> Gracias @estudiosistemas por los comentarios!, ya estarian los ajustes (tambien pueden usar directamente la herramenienta de sugerencia para dejar una modificacion a los archivos)
> 
> Sobre el tema de pyqr, podrias probar ejecutando `pyqr.exe --register`? , deberia funcionarles luego. Si el problema persiste, sugiero abrir otro ticket, porque no ha cambiado el QR y es un tema complementario (pueden seguir usando la version anterior).
> 
> Ya actualice todo el flujo de GiHub Actions para compilar, probar y generar el instalador automaticamente; pueden probar el que se prepara en los artefactos: https://github.com/reingart/pyafipws/pull/120/checks ![image](https://private-user-images.githubusercontent.com/1041385/418387996-f904bb5e-c15b-4e6d-a468-3e149eb5f553.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEyNTkxNjAsIm5iZiI6MTc0MTI1ODg2MCwicGF0aCI6Ii8xMDQxMzg1LzQxODM4Nzk5Ni1mOTA0YmI1ZS1jMTViLTRlNmQtYTQ2OC0zZTE0OWViNWY1NTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMDZUMTEwMTAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDJlNGMzOGU1MmZkYmVjZjM0NDQzYmY1YTljNDg1MGQ3NGY0Y2MxZTU5ZDU0NzUwNWI3NDk1YTRkNGE2M2FiMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0mYQ4tZlGhyyLaNVQ8pObZhz4uylc4aDhL3cHxXJyAM)

dcrocetti commented:
Buen día Grupo!

- Bajé los archivos que te comente antes;
- Usé los siguientes requirements
pytest==4.6.11; python_version <= '2.7'
pytest==7.2.0; python_version > '3'
pytest-html==1.22.1; python_version <= '2.7'
pytest-html==3.2.0; python_version > '3'
pytest-vcr==1.0.2; python_version <= '2.7'
pytest-cov==2.12.1; python_version <= '2.7'
pytest-freezegun==0.4.2; python_version <= '2.7'
pytest-vcr==1.0.2; python_version > '3'
pytest-cov==4.0.0; python_version > '3'
pytest-freezegun==0.4.2; python_version > '3'
pytest-mock==2.0.0; python_version <= '2.7'
pytest-mock==3.10.0; python_version > '2.7'
pywin32==304; sys_platform == "win32" and python_version > '3'
py2exe==0.11.1.1; sys_platform == "win32" and python_version > '3'

Tuve algún tema de identacion al principio con pyfepdf que lo comenté acá mismo, pero ya está resuelto en estos últimos archivos. 

No salió todo funcionando de una. Tenía errores como que no encontraba los módulos también. Lo que hice fue limpiar la carpeta dist y build y se solucionó ese tema con ese movimiento. 

- Para generar el instalador me daba error porque me pedía el vcredist.exe. Lo descargué y lo copie a la carpeta donde están todos los archivos .py.

Espero sirva la ayuda.

reingart commented:
Gracias @dcrocetti  por los comentarios y por confirmar que funciona.

Sugeriria si pueden abrir otro ticket para la actualizacion de las dependencias y el tema de vcredist.

Igualmente los instaladores se estan generando aca en GitHub (ver actions, se pueden descargar de ahi como comente antes).
Voy a revisar para ver que se publiquen automaticamente.

edyfe commented on wsfev1.py:
me da error l usar wsfev1 actaulizado en marzo 2025.en  ok = WSFEv1.EstablecerCampoFactura("cancela_misma_moneda_ext", N)

edyfe replied:
ya lo solucione. enviandolo DESPUES de crearfactura
mi duda es si antes del 15 de abril hay que mandarlo en N o no mandarlo
no aclaran bien si Cancela en la misma moneda de la factura o sea en Pesos va N
"N" ?

consertec commented:
Probaste pasarlo dentro de los parámetros del CrearFactura? Junto con caea,
fecha, cancela misma moneda y condición IVA receptor id.
Ah y creo que el "N" debe ir entre comillas.

El mar, 18 de mar de 2025, 4:42 p. m., edyfe ***@***.***>
escribió:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On wsfev1.py
> <https://github.com/reingart/pyafipws/pull/120#discussion_r2001869744>:
>
> me da error l usar wsfev1 actaulizado en marzo 2025.en ok =
> WSFEv1.EstablecerCampoFactura("cancela_misma_moneda_ext", N)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reingart/pyafipws/pull/120#pullrequestreview-2696045136>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEGXKWIQFGBVBXEYCZYMXT2VBZLTAVCNFSM6AAAAABXVIN25OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMOJWGA2DKMJTGY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

tiu80 commented:
buenas tardes, ya se encuentra el instalador de producción, yo estuve probando en homologación, pero no tuve suerte
